subsys/net/lib/openthread/platform/uart.c

```diff
@@ -208,7 +210,14 @@ otError otPlatUartSend(const uint8_t *aBuf, uint16_t aBufLength)
 	size_t cnt = ring_buf_put(ot_uart.tx_ringbuf, aBuf, aBufLength);
```

Comment:
not that ring buffer wraps and it may returns `cnt < aBufLength` on buffer boundary. If `cnt < aBufLength` then another attempt to write to ring buffer should be performed. Didn't you see occassional traffic loss?

Reply:
Actually it's not my piece of code, but no I didn't. Are you sure of this? I analyzed most use cases of this method in the whole project and it always goes similar to this. Just by using ring_buf_put() and at most logging error in case of returning less bytes than it was passed. Moreover please note that there is already a change (#26979) removing this line waiting for merge.

Reply:
> Are you sure of this?

yes. It means that `cnt` may be smaller than `aBufLength` but there is still available space in the ring buffer. That will lead to increased amount of data being dropped. Since system accepts that, it may go unnoticed.